android/app/src/main/java/org/fossasia/openevent/fragments/DayScheduleFragment.java

```diff
@@ -229,7 +232,26 @@ public void onClick(View view) {
 
     private void refresh() {
         if (NetworkUtils.haveNetworkConnection(getContext())) {
-            DataDownloadManager.getInstance().downloadSession();
+            if (NetworkUtils.isActiveInternetPresent()) {
+                //Internet is working
+                DataDownloadManager.getInstance().downloadSession();
+            } else {
+                //Device is connected to WI-FI or Mobile Data but Internet is not working
+                //show snackbar
+                //will be useful if user have blocked notification for this app
+                snackbar = Snackbar.make(getView(), R.string.waiting_for_network, Snackbar.LENGTH_INDEFINITE).setAction(R.string.snackbar_refresh_action,
```

Comment:
Better move this piece of code to a separate method as you are calling it multiple times in the code.